Rpackage/R/mmit.R

```diff
@@ -0,0 +1,66 @@
+#' The Max Margin Interval Tree
+#'
+#' Learning a regression tree for censored data.
+#'
+#' @param target.mat The response variable of the model
+#' @param feature.mat a data frame containing the feature variables in the model.
+#' @param margin margin paramaters 
+#' @param loss The type of loss; (\code{"hinge"}, \code{"square"})
+#' @param maxdepth The maximum depth criteia
+#' @param min_sample The minimum number of sample required 
+#' 
+#' @return The learned regression tree as an object of class party.
+#' 
+#' @author Toby Dylan Hocking, Alexandre Drouin, Torsten Hothorn, Parismita Das
+#' 
+#' @examples
+#' library(mmit)
+#' target.mat <- rbind(
+#'   c(0,1), c(0,1), c(0,1),
+#'   c(2,3), c(2,3), c(2,3))
+#' 
+#' feature.mat <- rbind(
+#'   c(1,0,0), c(1,1,0), c(1,2,0),
+#'   c(1,3,0), c(1,4,0), c(1,5,0))
+#' 
+#' colnames(feature.mat) <- c("a", "b", "c")
+#' feature.mat <- data.frame(feature.mat)
+#' 
+#' 
+#' out <- mmit(target.mat, feature.mat)
+#' 
+#' @export
+mmit <- structure(function(target.mat, feature.mat, weights = NULL,
+                           depth=0, maxdepth = Inf, margin=0.0, loss="hinge",
+                           id = 1L, min_sample = 1) {
```

Comment:
What is this `id` argument? It's not in the documentation and I don't think it's required.

Reply:
Some think for `weights`. There is not such argument in the python implementation.

Reply:
weights is just the name of array passed which determines which data to keep

Reply:
id is required for creation of partynode

Reply:
Yes, but these should not be specified by the user of the mmit function. They can be initialized to some value internally.

Reply:
shall i create a variable id inside mmit function and remove it from arguements?

Reply:
same with depth =0